lib/VMwareWebService/MiqVimVm.rb

```diff
@@ -720,8 +720,15 @@ def resetCd
 	#    In this case, backingFile must be the path to the existing VMDK.
 	# If backingFile is just the datastore name, "[storage 1]" for example,
 	#    file names will be generated as appropriate.
+	# 
+	# thinProvisioned - if new disk is thin (grow on demand, allowing storage overcommitment) or thick (pre-allocated)
+	# dependent       - if new disk is dependent (usual one) or not (meaning no delta file and no live snapshots of running VM)
+	# persistent      - if new disk should really save changes or discard them on VM poweroff
+	# 
+	# P.S. Good overview of use cases for dependent / independent and persistent / nonpersistent disks
+	# can be found at http://cormachogan.com/2013/04/16/what-are-dependent-independent-disks-persistent-and-non-persisent-modes/
 	#
-	def addDisk(backingFile, sizeInMB, label=nil, summary=nil, thinProvisioned=false)
+	def addDisk(backingFile, sizeInMB, label=nil, summary=nil, thinProvisioned=false, dependent=false, persistent=true)
```

Comment:
I meant the definition should change as follows:
```ruby
def addDisk(backingFile, sizeInMB, label=nil, summary=nil, options={})
```
Then, it could then be called as needed:
```ruby
addDisk(backing_file, d_size, persistent: false)
addDisk(backing_file, d_size, dependent: true)
addDisk(backing_file, d_size, thin_provisioned: true)
addDisk(backing_file, d_size, persistent: false, thin_provisioned: true, persistent: false)
```
Given `persistent` doesn't have a `false` or `nil` default value, you'll probably have to do something like:
```ruby
options = {:persistent => true}.merge(options)
```
in the function.